File: Linked_List/3217.Delete_Nodes_From_Linked_List_Present_in_Array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        Set<Integer> valuesToRemove = new HashSet<>();
        for (int num: nums){
            valuesToRemove.add(num);
        }

        ListNode dummy = new ListNode();
        dummy.next = head;
        ListNode res = dummy;
        while (dummy.next != null){
            if (valuesToRemove.contains(dummy.next.val)){
                ListNode tmp = dummy.next.next;
                dummy.next.next = null;
                dummy.next = tmp;
            } else {
                dummy = dummy.next;
            }          
        }
        return res.next;
    }
}

File: Linked_List/3217.Delete_Nodes_From_Linked_List_Present_in_Array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        nums_set = set(nums)
        dummy = ListNode()
        dummy.next = head
        res = dummy
        while dummy.next:

            if dummy.next.val in nums_set:
                tmp = dummy.next.next
                dummy.next.next = None
                dummy.next = tmp
            else:
                dummy = dummy.next

        return res.next
